### ISSUE TITLE
libRocket: too long strings displaces table elements

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#541]
OK, the title is a joke, but server name and other user-modifiable values must scroll horizontally or being truncated if they are too long, in order to not displace other columns like that:

![server list bug](http://dl.illwieckz.net/u/thomas-debesse/bazar/unvanquished/bugs/unvanquished_2015-01-24_052359_001.png)

Permanent horizontal scrolling is annoying, but we can imagine to truncate the name by default, and to scroll the name on mouse over.

**Comment by DolceTriade** [comment]
Mixed feelings about this. Not sure if I want to bring back marquee...

**Comment by illwieckz** [comment]
The `<marquee>` was a joke, but it would be nice if you can truncate strings when they are too long. :+1:

**Comment by illwieckz** [comment]
Instead of scrolling the string on mouse over, you can display a tooltip containing the whole string on mouse over, and truncate the string in the table element (I don't know how libRocket works and if it supports that, but some CSS tricks can truncate the printed string without truncating the string itself).

It could be a good idea if each table cell contain a duplication of its value as tooltip.

**Comment by Viech** [comment]
Not each table cell should have a tooltip, but I agree that this would be a nice thing to have for lines that have excess length.

**Comment by DolceTriade** [comment]
Agreed. The bug itself is real. For now, i will think of a way to handle this. Scrollbars? Left arrow to view rest of string? We'll see.

**Comment by Viech** [comment]
Please no scrollbars. Tooltip or marquee on mouse over should do.

**Comment by illwieckz** [comment]
> Not each table cell should have a tooltip.

Each table cell `==` less code `==` less bugs. :stuck_out_tongue_winking_eye: 

That's only why I suggested that. :wink:

**Comment by Viech** [comment]
If we produce fewer bugs, will you produce fewer issues? :stuck_out_tongue:

**Comment by illwieckz** [comment]
Yes of course :no_mouth: , or else I will replace bug reports by feature requests. :smile:

**Comment by illwieckz** [comment]
See #732 from @slipher for a duplicate of this bug on the scoreboard case, he posted this screenshot:

[![screenshot](https://i.imgur.com/8nlqS9z.jpg)](https://imgur.com/8nlqS9z)

Note that if it would be cool to have a hover tooltip for truncated names on server browser, truncation is probably only required for scoreboard.

Note that I tried to add many "overflow: hidden" styles in many places but have not yet managed to solve it. @DolceTriade said this will need to be a librocket fix.

**Comment by DolceTriade** [comment]
Tooltip is hard because the scoreboard doesn't get any input from the game. It is "view only", like the rest of the HUD. We could make the scoreboard part of the UI Menu system instead of the HUD, which would make it more powerful. Other games make it "view only" until you right click, and then you can interact with the scoreboard.

**Comment by illwieckz** [comment]
I think it's very ok to have a view-only scoreboard, that's why truncating usernames sounds good enough to me. We can be free to make more complex and advanced things in the future, but getting proper username truncation in scoreboard is enough to mark this issue fixed for the scoreboard case.

Having proper server name truncation in server list is enough to mark this issue fixed for the server list case too. Any other enhancement (tooltip, hover marquee…) can be requested as an _enhancement request_ as a new ticket once this bug is fixed and this report closed.

So, to sum it up:

- This bug is about not having long cells displacing columns (for which truncation is a solution);
- Providing an alternative way to read long names is another issue (which is a feature request);
- Implementing truncation will close this bug.

**Comment by DolceTriade** [comment]
Fixed by https://github.com/Unvanquished/Unvanquished/commit/a5b71150f6edb5a4f2edbc9785c0087971857678
